modules/order/src/OrderTotalSummary.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Drupal\commerce_order;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+
+class OrderTotalSummary implements OrderTotalSummaryInterface {
+
+  /**
+   * The adjustment type manager.
+   *
+   * @var \Drupal\commerce_order\AdjustmentTypeManager
+   */
+  protected $adjustmentTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(AdjustmentTypeManager $adjustment_type_manager) {
+    $this->adjustmentTypeManager = $adjustment_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildTotals(OrderInterface $order) {
+    $total = $order->getTotalPrice();
+    $subtotal = $order->getSubtotalPrice();
+    $collected_adjustments = $order->collectAdjustments();
+    $types = $this->adjustmentTypeManager->getDefinitions();
```

Comment:
We can remove the definitions from collectAdjustment(), question was where to filter out adjustments with invalid types. If we want to make client classes responsible for it, that's fine.